### PR TITLE
IBX-7653: Add FilterParser and IsContainer criterion parser

### PR DESCRIPTION
**Pull request by tischsoic**
| :ticket: Issue | IBX-7653 |
|----------------|-----------|

#### Related PRs: 
- https://github.com/ibexa/core/pull/333

### CODE REVIEW

**Comment by tischsoic** [comment]
Depends on other PR, regression run in: https://github.com/ibexa/commerce/pull/701

File: src/lib/Server/Input/Parser/Criterion/IsContainer.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Ibexa\Rest\Server\Input\Parser\Criterion;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\IsContainer as IsContainerCriterion;
+use Ibexa\Contracts\Rest\Exceptions;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use Ibexa\Rest\Input\BaseParser;
+use Ibexa\Rest\Input\ParserTools;
+
+class IsContainer extends BaseParser
```
Comment (Steveb-p) [inline src/lib/Server/Input/Parser/Criterion/IsContainer.php:17]:
```suggestion
final class IsContainer extends BaseParser
```

**Review by Steveb-p — APPROVED** [review]

File: src/lib/Server/Input/Parser/Criterion/IsContainer.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /** @var \Ibexa\Rest\Input\ParserTools */
+    protected $parserTools;
```
Comment (konradoboza) [inline src/lib/Server/Input/Parser/Criterion/IsContainer.php:20]:
```suggestion
    protected ParserTools $parserTools;
```

**Review by konradoboza — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by alongosz — APPROVED** [review]

**Comment by tischsoic** [comment]
CI is failing because [TMP] commits have been removed.

**Pull request merged**